src/dag/write.rs

```diff
@@ -43,22 +44,25 @@ impl<'a> Write<'_> {
     }
 
     pub async fn put_chunk(&mut self, c: &Chunk) -> Result<()> {
-        // TODO: These can be done in parallel.
-
-        self.kvw
-            .put(&Key::ChunkData(c.hash()).to_string(), c.data())
-            .await?;
-
-        if let Some(meta) = c.meta() {
-            self.kvw
-                .put(&Key::ChunkMeta(c.hash()).to_string(), meta)
-                .await?;
-        }
-
-        self.mutated_chunks
-            .write()
-            .await
-            .insert(c.hash().to_string());
+        let data_key = Key::ChunkData(c.hash()).to_string();
+        try_join!(
```

Comment:
I'm shocked you got this to work. Nice.

Reply:
The thing that made it easier was to extract the conditional one into a function at first... then I realized async blocks allows a concise syntax